photutils/psf/photometry.py

```diff
@@ -0,0 +1,427 @@
+            ``xcentroid`` and ``ycentroid`` are center position estimates of
+            the sources and ``flux`` contains flux estimates of the sources.
+            See, e.g., `~photutils.detection.DAOStarFinder`
+        group : callable or instance of any GroupStarsBase subclasses
+            ``group`` should be able to decide whether a given star overlaps
+            with any other and label them as beloging to the same group.
+            ``group`` receives as input an `~astropy.table.Table` object
+            with columns named as ``id``, ``x_0``, ``y_0``, in which ``x_0``
+            and ``y_0`` have the same meaning of ``xcentroid`` and
+            ``ycentroid``. This callable must return an `~astropy.table.Table`
+            with columns ``id``, ``x_0``, ``y_0``, and ``group_id``. The
+            column ``group_id`` should cotain integers starting from ``1``
+            that indicate in which group a given source belongs to.
+            See, e.g., `~photutils.psf.DAOGroup`
+        bkg : callable or instance of any `~photutils.BackgroundBase`
+            subclasses ``bkg`` should be able to compute either a scalar
+            background or a 2D background of a given 2D image.
+            See, e.g., `~photutils.background.MedianBackground`
+        psf : `astropy.modeling.Fittable2DModel` instance
+            PSF or PRF model to fit the data. Could be one of the models in
+            this package like `~photutils.psf.sandbox.DiscretePRF`,
+            `~photutils.psf.IntegratedGaussianPRF`, or any other suitable
+            2D model.
+            This object needs to identify three parameters (position of
+            center in x and y coordinates and the flux) in order to set them
+            to suitable starting values for each fit. The names of these
+            parameters should be given as ``x_0``, ``y_0`` and ``flux``.
+
+            `~photutils.psf.prepare_psf_model` can be used to prepare any 2D
+            model to match this assumption.
+        fitshape : array-like
+            Rectangular shape around the center of a star which will be used
+            to collect the data to do the fitting, e.g. (5, 5) means to take
+            the following relative pixel positions: [-2, -1, 0, 1, 2].
+            Also, each element of ``fitshape`` must be an odd number.
+        fitter : Fitter instance (default=LevMarLSQFitter())
+            Fitter object used to compute the optimized centroid positions
+            and/or flux of the identified sources. See
+            `~astropy.modeling.fitting` for more details on fitters.
+        niters : int (default=3)
+            Number of iterations to perform the loop FIND, GROUP, SUBTRACT,
+            NSTAR.
+        aperture_radius : float (default=None)
+            The radius (in units of pixels) used to compute initial estimates
+            for the fluxes of sources. If ``None``, one fwhm will be used. 
```

Comment:
Mmm. What happens if there is no FWHM? Gaussian PSFs have a FWHM, but `psf` could be an arbitrary model that does not have this parameter.